ledger/complete/compactor.go

```diff
@@ -247,6 +250,24 @@ Loop:
 				c.logger.Info().Msgf("compactor delayed checkpoint %d because prior checkpointing is ongoing", nextCheckpointNum)
 				nextCheckpointNum = activeSegmentNum
 			}
+
+		case _, ok := <-c.triggerCheckpointOnNextSegmentFinishCh:
```

Comment:
I think it would be better to move it to line 227? Would be clearer than having a nextCheckpoint setup here and wait for another loop